mobile_config/src/gateway_info_v3.rs

```diff
@@ -0,0 +1,282 @@
+        };
+
+        query
+            .fetch(db)
+            .map_err(anyhow::Error::from)
+            .try_fold(
+                MobileTrackerInfoMap::new(),
+                |mut map: MobileTrackerInfoMap, row| async move {
+                    let entity_key_b = row.get::<&[u8], &str>("entity_key");
+                    let entity_key = bs58::encode(entity_key_b).into_string();
+                    let last_changed_at = row.get::<DateTime<Utc>, &str>("last_changed_at");
+                    let asserted_location_changed_at =
+                        row.get::<Option<DateTime<Utc>>, &str>("asserted_location_changed_at");
+                    let asserted_location = row.get::<Option<i64>, &str>("asserted_location");
+
+                    map.insert(
+                        PublicKeyBinary::from_str(&entity_key)?,
+                        MobileTrackerInfo {
+                            location: asserted_location.map(|v| v as u64),
+                            last_changed_at,
+                            asserted_location_changed_at,
+                        },
+                    );
+                    Ok(map)
+                },
+            )
+            .await
+    }
+
+    // Streams all gateway info records, optionally filtering by device types.
+    pub fn all_info_stream_v3<'a>(
+        db: impl PgExecutor<'a> + 'a,
+        device_types: &'a [DeviceTypeV2],
+        mtim: &'a MobileTrackerInfoMap,
+    ) -> impl Stream<Item = GatewayInfoV3> + 'a {
+        // Choose base query depending on whether filtering is needed.
+        let query = if device_types.is_empty() {
+            sqlx::query(GET_MOBILE_HOTSPOT_INFO)
+        } else {
+            sqlx::query(&DEVICE_TYPES_METADATA_SQL).bind(
+                device_types
+                    .iter()
+                    // The device_types field has a jsonb type but is being used as a string,
+                    // which forces us to add quotes.
+                    .map(|v| format!("\"{}\"", v.to_sql_param()))
+                    .collect::<Vec<_>>(),
+            )
+        };
+
+        query
+            .fetch(db)
+            .filter_map(move |result| async move {
+                match result {
+                    Ok(row) => process_row(row, mtim).await,
+                    Err(e) => {
+                        tracing::error!("SQLx fetch error: {e:?}");
+                        None
+                    }
+                }
+            })
+            .boxed()
+    }
+
+    // Processes a single database row into a GatewayInfoV3, returning None if any step fails.
+    async fn process_row(
+        row: sqlx::postgres::PgRow,
+        mtim: &MobileTrackerInfoMap,
+    ) -> Option<GatewayInfoV3> {
+        let device_type = DeviceTypeV2::from_str(
+            row.get::<Json<String>, &str>("device_type")
+                .to_string()
+                .as_ref(),
+        )
+        .map_err(|err| sqlx::Error::Decode(Box::new(err)))
```

Comment:
Does we need to map this error? 

`.ok()?` will eat it and return None if it happens.